spring-web/src/main/java/org/springframework/http/RequestEntity.java

```diff
@@ -241,6 +284,17 @@ public static BodyBuilder method(HttpMethod method, URI url) {
 		return new DefaultBodyBuilder(method, url);
 	}
 
+	/**
+	 * Create a builder with the given method and given string base uri template.
+	 * @param method the HTTP method (GET, POST, etc)
+	 * @param uri the uri
+	 * @return the created builder
+	 */
+	public static BodyBuilder method(HttpMethod method, String uri) {
+		return new DefaultBodyBuilder(method, uri);
+	}
```

Comment:
All methods that take a `String` uri need to mention in the Javadoc that this is a URI template that may contain URI variable placeholders and mention that URI variables can be provided through the builder either as a vararg or as a map (probably best to `@link` to the corresponding builder methods). 

It be useful also to link to `getUrl()` which would expand through `DefaultUriBuilderFactory` and encode as explained in `DefaultUriBuilderFactory#setEncodingMode`, unless the `UriTemplateHandler` is provided.

To avoid repeating this in many places, you could document at the class level that there are `URI` and `String` uri based factory methods, and explain how the `String` uri variants work. Then refer to class-level Javadoc from each factory method.

Reply:
Sorry for pause. I had a rest.
I added javadocs, but maybe also added grammatical errors, because I'm not a native English speaker

Reply:
Okay thanks.